### PR TITLE
fix: XYNE-63982 Skip non-meeting Google Calendar events at ingestion

### PR DESCRIPTION
**Pull request by sayantanp-eng**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
XYNE-63982

Google Calendar syncs workingLocation/outOfOffice/focusTime/birthday events (Google-native, non-meeting event types) into the `calls` table as real Call rows, so they render as full-height pills in the calendar sidebar/week
view. `GCalEvent.eventType` was parsed off the wire but never read anywhere in the ingestion path.

Fix: `isNonMeetingEvent(event)` — allow-list on `eventType === 'default'`/unset — added as an early return in `storeGCalEventAsCall`, and applied to the `fetchedExternalIds` set in `storeGCalEventsAsCallsForUser` so any row already polluted before this deploy gets cancelled by the existing "remove what's no longer fetched" path on the next full/manual sync, instead of being silently kept forever.

Bundled in the same commit (pre-existing local changes, not re-split):
- `useCallHistory(user?.id, { isCalendarView: true })` in `ActivityCalendarWeekView.tsx` / `XyneCalendarSidebar.tsx` — the calendar week/day views were only loading the next-2-days upcoming-calls query, so a scheduled call further out never appeared in the calendar grid at all.
- `CallDetailSidebarView.tsx` — "View in Google Calendar"/"View in Microsoft Outlook" restyled from a small text link to a full-width button matching the Join Call button's styling.

## Areas Touched

- [x] `apps/backend` (API / worker)
- [x] `apps/dashboard`
- [ ] `apps/electron`
- [ ] `apps/xyne-claw` / `apps/xyne-claw-auth`
- [ ] `packages/shared` or another shared package
- [ ] Infra (docker, scripts, nix, CI)

---

## Zero (Sync) Changes

- [x] No Zero changes — **skip this section**
- [ ] Adds/modifies a **mutator**
- [ ] Adds/modifies the **schema** (tables, columns, relationships)
- [ ] Adds/modifies a **query or ACL**

## Schema & Data Changes

- [x] No schema changes — **skip this section**
- [ ] Prisma schema modified (`apps/backend/prisma` and/or `prisma-common`)
- [ ] Migration added
- [ ] Backfill / data migration included

## Config, Environment & URLs

- [x] No config changes — **skip this section**
- [ ] Env variable added / renamed / removed
- [ ] **Base URL, host, port, or origin changed**
- [ ] CORS / allowed origins / OAuth redirect URIs changed
- [ ] Feature flag or runtime config changed

## API Contract

- [x] No API changes
- [ ] New endpoint
- [ ] Existing contract modified (shape, status codes, auth)
- [ ] Breaking for dashboard / electron / external / bots

---

## How did you test it?
Backend: `apps/backend/scripts/seed-gcal-nonmeeting-events.ts` — feeds
`storeGCalEventAsCall` synthetic workingLocation/outOfOffice/focusTime/
birthday payloads through the real ingestion path against the local dev DB;
confirmed the guard skips all four (`--clear` variant confirms cleanup of
pre-existing rows via the same predicate). No dashboard/browser check was
done for the `isCalendarView` or button-restyle changes in this pass —
worth a manual look before merge.

### Automated

- [ ] Backend unit tests — `pnpm --filter xyne-spaces-backend run test`
- [ ] Claw tests — `pnpm --filter xyne-claw run test`
- [ ] End-to-end suite — `pnpm run test`
- [ ] Added / updated tests for this change
- [x] Not covered by tests 

### Manual

**Users**
- [x] Single user 
- [ ] Two users at once
- [ ] Different roles or permission levels
- [ ] Different workspaces / spaces

**Login**
- [ ] Google
- [ ] Microsoft
- [ ] Dev auth (`ENABLE_DEV_AUTH`)
- [ ] Fresh signup / first-time user
- [ ] Session expiry, re-login, logout

**Run mode**
- [x] Local dev (`pnpm run dev:all`) 
- [ ] Test mode (`dev:test`)
- [ ] Sandbox / claw test (`claw:test`)
- [ ] Prod-like local (`dev:prod`)
- [ ] Docker compose
- [ ] Electron desktop

**Sync & resilience**
- [ ] Multiple tabs stay consistent
- [ ] Reload / hard refresh
- [ ] Offline then reconnect
- [ ] Concurrent edits on the same entity

**Surface & data**
- [ ] Web browser 
- [ ] Electron desktop
- [ ] Narrow viewport, dark + light theme
- [ ] Empty state
- [ ] Large / realistic data volume
- [ ] Error paths

---

## Checklist

- [x] Reviewed my own diff; scoped to one thing 
- [ ] `pnpm run build:shared` passes 
- [x] Backend `typecheck` passes (`npx tsc --noEmit`); full `build` not run
- [x] Dashboard `typecheck` passes (`npx tsc --noEmit`); `lint:errors-only` + `build` not run
- [x] Formatting clean in the packages I touched
- [ ] New deps added — n/a, none added
- [x] Commits follow `<type>: <TICKET-ID> <subject>`; branch is `fix/*`
- [ ] Docs / guidelines updated — n/a, no user-facing docs for this
- [x] No debug logging or commented-out code left behind

### CODE REVIEW

**Review by Mayankbansal-juspay — APPROVED** [review]

**Review by samitbarai — APPROVED** [review]

File: apps/backend/src/services/googleCalendarCallStore.ts [hunk 1]
```diff
@@ -207,7 +213,9 @@ export async function storeGCalEventsAsCallsForUser(
   if (isFullSync && !options?.skipCancelRemoved && !hitStoreCap) {
     const externalIdPrefix = buildCalendarExternalIdPrefix('google', userId);
     const fetchedExternalIds = new Set(
-      eventsToStore.filter((e) => e.id).map((e) => buildCalendarExternalId('google', userId, e.id!))
+      eventsToStore
+        .filter((e) => e.id && !isNonMeetingEvent(e))
```
Comment (XyneSpaces) [inline apps/backend/src/services/googleCalendarCallStore.ts]: **[major | R2 + guideline #233]** — The PR description claims any row polluted before this deploy "gets cancelled by the existing remove-what's-no-longer-fetched path on the next full/manual sync" — that doesn't hold for **past-dated rows**. The cancel pass only reconciles rows inside the sync window: `performManualSync` passes `timeRange: { startsAfter: now, startsBefore: now + 30d }` (`apps/backend/src/queues/googleCalendarSyncQueue.ts:113-130`) and `findExternalCalendarCalls` applies it as `startsAt >= now` (`callRepository.ts:2330`), so a polluted row whose `startsAt` is already in the past (e.g. the Mon Sep 21 "Office" row behind XYNE-63982) is never a cancellation candidate — only a syncToken expiry (baseline path, no timeRange) or a manual DB update clears it. *(Cross-file behavior verified by the attached RCA report — not visible in this diff.)*

↳ FIX: in the follow-up `fix/` PR, drop `startsAfter` from the manual-sync cleanup window (or set it to `now - 30d`) so past rows become reconcilable.

File: apps/backend/src/services/googleCalendarCallStore.ts [hunk 2]
```diff
@@ -207,7 +213,9 @@ export async function storeGCalEventsAsCallsForUser(
   if (isFullSync && !options?.skipCancelRemoved && !hitStoreCap) {
```
Comment (XyneSpaces) [inline apps/backend/src/services/googleCalendarCallStore.ts]: **[major | R2 + R6]** — The new `!isNonMeetingEvent(e)` exclusion only takes effect when this cancel pass runs, but the pass is skipped on truncation: when the Google fetch truncates at `MAX_CALENDAR_EVENTS_PER_SYNC = 100` (`apps/backend/src/services/calendarSyncConfig.ts`) the caller sets `skipCancelRemoved`, and `hitStoreCap` skips it here too — so for calendars with >100 events in the 30-day window, cleanup silently never runs and pre-existing pollution becomes permanent. Nothing logs or metrics the skip. *(Cross-file behavior verified by the attached RCA report.)*

↳ FIX: decouple the cancel pass from the store cap (or run it against the full fetched external-id set), and emit a warn-level log/metric whenever the cancel pass is skipped so the silent stop is observable.

File: apps/dashboard/src/components/Activity/ActivityCalendarWeekView.tsx [hunk 3]
```diff
@@ -64,7 +64,7 @@ export const ActivityCalendarWeekView = (): ReactElement => {
     deleteModalCall,
     handleDeleteConfirm,
     closeDeleteModal,
-  } = useCallHistory(user?.id);
+  } = useCallHistory(user?.id, { isCalendarView: true });
```
Comment (XyneSpaces) [inline apps/dashboard/src/components/Activity/ActivityCalendarWeekView.tsx]: **[major | R2 + guidelines #204/#208]** — This bundled option-flip (same change at `XyneCalendarSidebar.tsx:1102`) shipped with no browser check — the PR's own testing note says "No dashboard/browser check was done for the `isCalendarView` … changes … worth a manual look before merge". It also extended the #1974-introduced stale-cache fallback to these views: after one visit to Calendar view, the disabled calendar query still returns its IndexedDB-persisted cache, and the `scheduledCallRows ?? upcomingScheduledCallRows` fallback in `useCallHistory` let that frozen snapshot shadow the live upcoming-calls query — resurrecting ended calls as joinable rows ("scheduled calls repeating again and again"), fixed in #2198 (merged 2026-09-22, targeted at `release-20260921`).

↳ FIX: carry a regression test for the calendar→list view transition in the follow-up, and keep behavior-changing frontend flips out of backend fix PRs (#264/#291).

File: apps/backend/src/services/googleCalendarCallStore.ts [hunk 4]
```diff
@@ -82,6 +82,10 @@ function parseGCalDateTime(dt?: GCalDateTime): Date | undefined {
   return isNaN(d.getTime()) ? undefined : d;
 }
 
+function isNonMeetingEvent(event: GCalEvent): boolean {
+  return !!event.eventType && event.eventType !== 'default';
```
Comment (XyneSpaces) [inline apps/backend/src/services/googleCalendarCallStore.ts]: **[major | R7 + guideline #204]** — No automated coverage for the new predicate or the `fetchedExternalIds` exclusion — the checklist marks "Not covered by tests" and the seed script (`seed-gcal-nonmeeting-events.ts`) is a manual repro tool, not a test. The edge cases are exactly what needs pinning: unset `eventType` (older wire data) must pass through, unknown future Google eventTypes get skipped, and a pre-polluted row must actually flip to `CANCELLED` via the exclusion.

↳ FIX: unit tests for `isNonMeetingEvent` (unset / `default` / `workingLocation` / `outOfOffice` / `focusTime` / `birthday` / unknown) plus a test asserting the cancel path removes a pre-existing non-meeting row.

File: apps/dashboard/src/components/Chat/XyneCalendarSidebar/CallDetailSidebarView.tsx [hunk 5]
```diff
@@ -452,10 +452,10 @@ function CallLocationSection({
           rel='noopener noreferrer'
           data-track-category='Calendar'
           data-track-name='CALL_DETAIL_OPEN_EXTERNAL'
-          className='mt-2 inline-flex items-center gap-1.5 text-xs font-medium text-muted-foreground hover:text-foreground'
+          className='mt-2 flex h-9 w-full items-center justify-center gap-2 rounded-lg bg-foreground text-sm font-semibold text-background hover:bg-foreground/90'
```
Comment (XyneSpaces) [inline apps/dashboard/src/components/Chat/XyneCalendarSidebar/CallDetailSidebarView.tsx]: **[minor | R9 + guideline #264]** — Unrelated UI restyle (`ExternalLink`→`LinkSlant`, full-width button styling) bundled into an ingestion-guard fix; the PR admits these are "pre-existing local changes, not re-split". Bundling is what let the `isCalendarView` regression above ship without a browser check.

↳ FIX: move drive-by UI changes to their own PR (#264/#291).

**Review by XyneSpaces — COMMENTED** [review]
Xyne Spaces Reviewer — post-merge review of PR #2064, triggered by the XYNE-63982 RCA. One inline comment per finding; verdict, deployment context, and reviewer recommendation are in the top-level summary comment. No blocking issues — the 4 major findings target the recommended follow-up `fix/` PR.

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #2064

**Verdict:** Changes requested — no blockers, but 4 major findings need the follow-up `fix/` PR. (PR already merged 2026-09-18; reviewed post-merge off the back of the XYNE-63982 RCA. Per that RCA, the *original* symptom additionally needs a `generic-worker` redeploy to `release-20260921`+ before this code runs in pre-prod at all.)
**Reviewed:** 4 files, 8 hunks · **Findings:** 0 blocking, 4 major, 1 minor, 1 nit

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| major | R2 · #233 | `apps/backend/src/services/googleCalendarCallStore.ts:217` | Cleanup claim ("polluted rows get cancelled on the next full/manual sync") doesn't hold for past-dated rows — the manual-sync cancel pass is window-scoped (`startsAfter: now` via `googleCalendarSyncQueue.ts:113-130` → `startsAt >= now` in `callRepository.ts:2330`), so a row that already started can never be cancelled by a sync (RCA-verified, cross-file). |
| major | R2 · R6 | `apps/backend/src/services/googleCalendarCallStore.ts:213` | The new exclusion only runs when the cancel pass runs — truncation at `MAX_CALENDAR_EVENTS_PER_SYNC = 100` sets `skipCancelRemoved` (and `hitStoreCap` skips too), so busy calendars silently never clean up (RCA-verified, cross-file). |
| major | R2 · #204/#208 | `apps/dashboard/src/components/Activity/ActivityCalendarWeekView.tsx:67` (+ `XyneCalendarSidebar.tsx:1102`) | Bundled `isCalendarView: true` flips shipped with no browser check and extended the #1974 stale-cache fallback — ended calls resurrected as joinable ("scheduled calls repeating"), fixed in #2198 (merged 2026-09-22 to `release-20260921`). |
| major | R7 · #204 | `apps/backend/src/services/googleCalendarCallStore.ts:86` | No automated tests for `isNonMeetingEvent` or the `fetchedExternalIds` exclusion (checklist: "Not covered by tests"; seed script is a manual repro tool). |
| minor | R9 · #264 | `apps/dashboard/src/components/Chat/XyneCalendarSidebar/CallDetailSidebarView.tsx:455` | Unrelated button restyle bundled into the ingestion-guard fix. |
| nit | R6 | `apps/backend/src/services/googleCalendarCallStore.ts:114` | Non-meeting skips are silent — no log/metric to triage "why didn't my calendar event sync". |

**The follow-up `fix/` PR should carry:** (1) drop/widen `startsAfter` for the cleanup window (or `now - 30d`), (2) warn-log `skipCancelRemoved` and decouple the cancel pass from the store cap, (3) unit tests for the predicate/exclusion plus a calendar→list-view regression test.

### Deployment context (from the attached RCA)
- Pre-prod `generic-worker` is on `1.323.0-release-20260916.8`; the fix first ships in `release-20260921`+. Redeploy the worker, trigger Sync Google Calendar, and look for `[STORE] Cancelled N removed event(s)`.
- Even after the redeploy, the Mon Sep 21 "Office" row survives until finding 1 is fixed.
- Unverified (RCA): if the "Office" entries are plain all-day events (`eventType: 'default'`) rather than Google-typed `workingLocation` events, this guard won't filter them — check how the event was created in Google Calendar.

### Recommended reviewer
**@Mayankbansal-juspay** — strongest familiarity with the Google Calendar sync backend, where this PR's primary change, all 4 major findings, and the follow-up live.

| Candidate | Owns (files in this PR) | Default reviewer? | Pending reviews | Tie-breaker |
|-----------|-------------------------|-------------------|-----------------|-------------|
| @Mayankbansal-juspay | `googleCalendarCallStore.ts` (lineage: authored the calendar-sync worker ingestion #1362, GCal push #1590; approved this PR at head SHA) — exact commits-by-path not verifiable this run, inferred from PR history | no — repo CODEOWNERS scopes only Zero query-ACL paths, none of these files | 339 open PRs request their review | deepest calendar-backend familiarity |
| @monika-annamaneni | `useCallHistory.ts` (blast-radius file — authored the #2198 fix for the regression these flips extended) | no | 0 | most recently active in this area (#2198 merged 2026-09-22); capacity alternative |
| @samitbarai | — (approved this PR; no file-ownership evidence found this run) | no | 367 | — |

Ranking is familiarity-first: the follow-up is backend calendar-sync work in @Mayankbansal-juspay's lineage, so he gets the nod despite the deepest queue (339 pending review requests; the numbers for both approvers look inflated by default-reviewer automation). If the follow-up leans frontend or his queue is the constraint, @monika-annamaneni (0 pending, authored yesterday's #2198 fix in this exact area) is the capacity pick. Signals not computable this run: commits-by-path file ownership (no tool available — inferred from PR history, labeled as such) and CODEOWNERS match (none covers these paths).

*Note: this automation run has no GitHub request-reviewers API or `gh` CLI access, so the review request is this @mention — @Mayankbansal-juspay please pick up the follow-up, or add yourself as reviewer on it.*

File: apps/backend/src/services/googleCalendarCallStore.ts [hunk 6]
```diff
@@ -107,6 +111,8 @@ export async function storeGCalEventAsCall(
 ): Promise<void> {
   if (!event.id) return;
 
+  if (isNonMeetingEvent(event)) return;
```
Comment (XyneSpaces) [inline apps/backend/src/services/googleCalendarCallStore.ts]: **[nit | R6]** — Non-meeting events are dropped silently — there is no log or metric when this guard fires, so "why didn't my calendar event sync?" is undebuggable from the `[STORE]` logs.

↳ FIX: a structured `logger.info`/counter (event id + eventType) when skipping — via the structured logger, not console.log (#240).
